openwhyd/openwhyd
# fix(tests): Reset database between each Cypress tests

Extracted from PR #301.

Contributes to #199.

## What does this PR do / solve?

Cypress' best practices recommend to start each test with a clean server state.

## Overview of changes

Instead of seeding the database before running Cypress, we let it do that before running every test.

As a result this PR:
- introduces a new `POST /testing/reset` API endpoint that clears the db and re-populates it with test data
- removes calls to `npm run test-reset`
- extracts the db initialization logic from `app.js` to `mongodb.js`
- also takes that opportunity to extract `cacheCollections()` and `runShellScript()` out of `mongodb.js`'s `init()` function

## How to test this PR?

CI should pass.

Locally, you should be able to run all tests more than once in a row, using `$ npm run docker:test:cypress:dev`
